specification/src/main/asciidoc/platform/ApplicationProgrammingInterface.adoc

```diff
@@ -304,43 +321,78 @@ _jakarta.xml.rpc.handler.MessageContext_ type.]
 
 
 [[a2331]]
-==== Pruned Java Technologies
+==== Optional Jakarta Technologies
 
 As the Jakarta EE specification has evolved,
 some of the technologies originally included in Jakarta EE are no longer as
 relevant as they were when they were introduced to the platform. The
-Jakarta EE expert group follows a process first defined by the Java SE
-expert group ( _http://mreinhold.org/blog/removing-features_ ) to prune
+Jakarta EE Platform Specification Project follows a process similar to the one first defined by the Java SE
+expert group ( _http://mreinhold.org/blog/removing-features_ ) to stabilize and deprecate
 technologies from the platform in a careful and orderly way that
 minimizes the impact to developers using these technologies, while
-allowing the platform to grow even stronger. In short, the process
-defines two steps:
+allowing the platform to grow even stronger. In short, our process
+defines three steps:
+
+. The Platform Specification Project for release N 
+of the platform decides to propose that a particular feature be
+marked "proposed optional". The specification for that release documents the proposal.
+. The Platform Specification Project for release N+1 decides whether to
+mark the feature as "optional" for this N+1 release, retain it as a required component,
+or leave it in the "proposed optional" state for the next N+2 release cycle to decide.
+. Eventually, the Platform Specification Project for release N+2 (or beyond) can decide
+to officially "prune" (remove) the "optional" feature from the Platform.
 
+The result of successfully applying this
+policy to a feature is to allow a gradual deprecation and possible removal
+of the feature from a required component of the platform.
+Product vendors can _choose_ to remove an "optional" feature from their products.
+Product vendors are _required_ to remove a "pruned" feature from their products for
```

Comment:
Another topic for the list, but this is likely too restrictive.  For example we've removed CORBA but left remote calls as optional.  GlassFish and WAS can only support remote calls via CORBA.  WAS could remove remote calls entirely, but GlassFish is not capable of doing that as we need at least one Compatible Implementation that supports all optional parts.   Therefore there's no way it can both remove CORBA *and* support the optional remote EJB calls.

There are other reasons that can make this hard.  We would likely need some compromise that would allow:

 - someone to continue to ship a compliant impl of something that may exist as a standalone specification.
 - the platform to freely add standalone specs to the platform definition (say JCache, NoSQL) and later remove them from the platform definition without "outlawing" its use by the industry in general.  There needs to be some way to return to a simple standalone spec status.

Reply:
Yep, looks like I overstepped the requirement for the "removed" features.  I'll remove or reword this statement.

Reply:
Reworded instead of complete removal of this statement.  I want to be clear that "removed" features are no longer maintained.  And, that any integration of a "removed" feature with the Platform is left as an exercise for the vendor.  That is, it is not defined by the Spec.

Reply:
I suggest to remove "Product vendors are _required_ to remove a "pruned" feature from their products for" as Product vendors can decide to remove a pruned feature or not.

Reply:
@Emily-Jiang, thanks. That comment was already addressed in the commits.